scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py

```diff
@@ -390,6 +434,37 @@ def _compile_s3_top_level_dir_name(self, dir_name: str):
             return dir_name
         raise SystemError(
             "s3_top_level_dir not set while trying to access s3 instance.")
+    
+    def _extract_test_runs_from_test_run_report(self, report: dict):
+        """
+        Extract all test runs from a test run report and store in one list
+        @param report: The test run report.
+        """
+        PASSING_TEST_RUNS = "passing_test_runs"
+        FAILING_TEST_RUNS = "failing_test_runs"
+        EXECUTION_FAIL_TEST_RUNS = "execution_failure_test_runs"
+        TIMED_OUT_TEST_RUNS = "timed_out_test_runs"
+        UNEXECUTED_TEST_RUNS = "unexecuted_test_runs"
+        test_runs = []
+        test_runs += report[PASSING_TEST_RUNS]
+        test_runs += report[FAILING_TEST_RUNS]
+        test_runs += report[EXECUTION_FAIL_TEST_RUNS]
+        test_runs += report[TIMED_OUT_TEST_RUNS]
+        test_runs += report[UNEXECUTED_TEST_RUNS]
+        return test_runs
+
+    def _extract_test_runs_from_sequence_report(self, report):
+        """
+        Extract all relevant test runs from the report generated by the TIAF runtime.
+        @param report: The report generated by the TIAF runtime, in dict format.
+        """
+        report_type = report['type']
+        test_runs = self._extract_test_runs_from_test_run_report(report['selected_test_run_report'])
```

Comment:
Can these string literals be replaced with named constants? They might already be in the MARS script.

Reply:
Done